Fix API documentation for change-rules endpoint

adrian-golian opened the pull request:
This PR updates the documentation for the /games/{game_uuid}/change-rules endpoint to reflect its actual behavior.

### What Changed
Corrected the documentation to remove references to a rules object in the request.
Updated to show that rule parameters are passed as individual query parameters, not as a JSON object.
Clarified that the endpoint uses the GET method, despite modifying state.
Added more detail on:
Required and optional query parameters
Error handling
Behavior of max_cards and time_limit
Sample curl calls
### Why
The original API documentation was incorrect — it referenced a rules object being passed in the request, which was never supported by the backend. This led to confusion during integration and testing.
Impact
No backend code changes.
Documentation now accurately reflects the existing behavior of the change-rules endpoint.
Prevents further misunderstandings or incorrect API usage.

adrian-golian committed:
Fix API documentation for change-rules endpoint

maciej-pomykala commented on api/README.md:
```diff
@@ -181,43 +181,69 @@ curl <HOST>/games/f2fdd601-bc82-438b-a4ee-a871dc35561a/set-readiness?player_uuid
+
+   **Optional (at least one rule parameter should be provided):**
+
+   * `"time_limit"=integer` (0 for no limit, or 6-300 seconds)
```
1-300 seconds, actually

maciej-pomykala commented on api/README.md:
```diff
@@ -181,43 +181,69 @@ curl <HOST>/games/f2fdd601-bc82-438b-a4ee-a871dc35561a/set-readiness?player_uuid
+
+  * Rule parameters are passed as **individual query parameters**, NOT as a nested JSON object
+  * Only the admin (creator) of the game can change rules
+  * Rules can only be changed before the game starts (status = NOT_STARTED)
```
In the language of the game state object returned by the API, it's spelt `Not started`

maciej-pomykala commented on api/README.md:
```diff
@@ -181,43 +181,69 @@ curl <HOST>/games/f2fdd601-bc82-438b-a4ee-a871dc35561a/set-readiness?player_uuid
+  * Rule parameters are passed as **individual query parameters**, NOT as a nested JSON object
+  * Only the admin (creator) of the game can change rules
+  * Rules can only be changed before the game starts (status = NOT_STARTED)
+  * When `time_limit` is set to non-zero, all human players' readiness is reset to false
```
I'd say this is misleading and a better explanation would look like: 
When `time_limit` is non-zero after any rule changes resulting from the current call to the endpoint have been made, all human players' readiness is reset to false

maciej-pomykala commented on api/README.md:
```diff
@@ -181,43 +181,69 @@ curl <HOST>/games/f2fdd601-bc82-438b-a4ee-a871dc35561a/set-readiness?player_uuid
+  * Only the admin (creator) of the game can change rules
+  * Rules can only be changed before the game starts (status = NOT_STARTED)
+  * When `time_limit` is set to non-zero, all human players' readiness is reset to false
+  * The `max_cards` preference may be overridden by the engine if not feasible for the player count
```
It might be better to emphasise that the `max_cards_preference` property of the rules is not overridden, it might just not be directly reflected in the `max_cards` property of the game state

maciej-pomykala reviewed (changes requested):
We need some corrections to the changed lines

adrian-golian committed:
Update change-rules endpoint docs for accuracy and clarity 

- Corrected time_limit range to 1–300 seconds
- Updated game status wording to match API ("Not started")
- Clarified when human readiness is reset after rule changes
- Reworded explanation of max_cards_preference vs max_cards behavior
